spectral_fitting/fitting_with_spectralcube.ipynb

```diff
@@ -0,0 +1,697 @@
+     },
+     "metadata": {
+      "needs_background": "light"
+     }
+    }
+   ],
+   "source": [
+    "plt.plot(spec.spectral_axis, spec - g_fit(spec.spectral_axis))\n",
+    "plt.ylabel(\"Brightness Temperature (K)\")\n",
+    "plt.xlabel(\"Radio Velocity (km /s)\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Some residuals remain, however, the fit is fairly good.\n",
+    "\n",
+    "One major feature of `astropy.modeling.models` is the built-in handling of units. The fitted model `g_fit` already has each parameter in useful physical units."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Exercise ###\n",
+    "\n",
+    "Can the above be example fit be improved with a more sophisticated model? Try the fit using multiple Gaussian components."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Using dask ###\n",
+    "\n",
+    "When using a large spectral-line data cube, the [dask](https://dask.org/) library can be used in spectral-cube for operations instead of numpy. See the [spectral-cube documentation](https://spectral-cube.readthedocs.io/en/latest/dask.html) on dask integration for a thorough description.\n",
+    "\n",
+    "Most of the above code for fitting a spectrum will be unchanged. What will change is how the data cube is read in:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cube = SpectralCube.read(datafile, use_dask=True, format='fits')\n",
+    "cube = cube.with_spectral_unit(u.km / u.s)\n",
+    "\n",
+    "cube.use_dask_scheduler('synchronous')\n",
+    "\n",
+    "from dask.diagnostics import ProgressBar\n",
```

Comment:
Maybe make the progress bar stuff a separate cell and explain what you are doing before each step rather than a posteriori below.

Reply:
Done